backend/penndata/views.py

```diff
@@ -104,7 +104,16 @@ class Events(generics.ListAPIView):
     serializer_class = EventSerializer
 
     def get_queryset(self):
-        return Event.objects.filter(event_type=self.kwargs.get("type", ""))
+        queryset = Event.objects.all()
+
+        event_type = self.kwargs.get("type")
+        if event_type:
+            queryset = queryset.filter(event_type=event_type)
+
+        queryset = queryset.filter(
+            end__gte=timezone.localtime(), start__lte=timezone.localtime() + timedelta(days=30)
```

Comment:
why does `start` have to be less than 30 days from now?

Reply:
events displayed should start within next 30 days, if they haven't started yet. if they have started already, then only show if they haven't ended yet